### PR TITLE
VRF-253 Check the latest block number in the transmit checker for V1

### PR DESCRIPTION
**Pull request by vreff**

### CODE REVIEW

File: core/chains/evm/txmgr/transmitchecker.go [hunk 1]
```diff
@@ -168,9 +180,36 @@ func (v *VRFV1Checker) Check(
...
+	if err != nil {
+		l.Errorw("Failed to fetch latest header. Attempting to transmit anyway.",
+			"err", err,
+			"ethTxID", tx.ID,
+			"meta", tx.Meta,
+		)
+		return nil
+	}
+
+	r, err := v.TransactionReceipt(ctx, *meta.RequestTxHash)
```
Comment (makramkd) [inline core/chains/evm/txmgr/transmitchecker.go:193]: I think we should check if `meta.RequestTxHash` is `nil` earlier in the func and `return nil` if it is indeed nil (i.e `Attempting to transmit anyway`)

File: core/chains/evm/txmgr/transmitchecker.go [hunk 2]
```diff
@@ -134,6 +139,13 @@ type VRFV1Checker struct {
 	// Callbacks checks whether a VRF V1 request has already been fulfilled on the VRFCoordinator
 	// Solidity contract
 	Callbacks func(opts *bind.CallOpts, reqID [32]byte) (v1.Callbacks, error)
+
+	// HeaderByNumber fetches the header given the number. If nil is provided,
+	// the latest header is fetched.
+	HeaderByNumber func(ctx context.Context, n *big.Int) (*gethtypes.Header, error)
```
Comment (makramkd) [inline core/chains/evm/txmgr/transmitchecker.go:145]: I think we should use `HeadByNumber` here since it's technically more correct - `HeaderByNumber` will return the wrong blockhash for e.g avalanche.

File: core/chains/evm/txmgr/transmitchecker.go (hunk 1 above)
Comment (makramkd) [inline core/chains/evm/txmgr/transmitchecker.go:193]: Possibly a micro-optimization: should we do `HeadByNumber` and `TransactionReceipt` as a `BatchCallContext` instead? Would save a round trip to the RPC node.

File: core/chains/evm/txmgr/transmitchecker_test.go [hunk 3]
```diff
@@ -217,6 +237,22 @@ func TestTransmitCheckers(t *testing.T) {
 			tx, attempt := newTx(t, r2)
 			require.NoError(t, checker.Check(ctx, log, tx, attempt))
 		})
+
```
Comment (makramkd) [inline core/chains/evm/txmgr/transmitchecker_test.go:240]: Also need a test case for `RequestTxHash` being `nil`.

**Review by makramkd — APPROVED** [review]
